web/client/actions/security.js

```diff
@@ -168,3 +173,50 @@ export function verifySession() {
 }
 
 export const checkLoggedUser = () => ({type: CHECK_LOGGED_USER});
+
+/**
+ * set status to show modal for inserting credentials
+ * @param {boolean} status
+ */
+export const setShowModalStatus = (status) => {
+    return {
+        status,
+        type: SET_SHOW_MODAL_STATUS
+    };
+};
+/**
+ * set protected services to request to provide username and password
+ * @param {object[]} protectedServices
+ */
+export const setProtectedServices = (protectedServices) => {
+    return {
+        protectedServices,
+        type: SET_PROTECTED_SERVICES
+    };
+};
+/**
+ * clear security of layers
+ * @param {string} id
+ */
+export const clearSecurity = (protectedId) => {
+    return {
+        protectedId,
+        type: CLEAR_SECURITY
+    };
+};
+/**
+ * set credentials for a service ogc
+ * @param {object} protectedService
+ */
+export const setCredentialsAction = (protectedService, creds) => {
+    if (creds && protectedService.protectedId) {
+        setCredentials(protectedService.protectedId, {
+            ...creds,
+            url: protectedService.url
+        });
+    }
+    return {
+        protectedService,
+        type: SET_CREDENTIALS
+    };
+};
```

Comment:
Not sure if it would be better to have the setCredentials side effect inside the reducer instead of the action